sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/quickpulse/filtering/DependencyDataColumns.java

```diff
@@ -0,0 +1,95 @@
+package com.azure.monitor.opentelemetry.autoconfigure.implementation.quickpulse.filtering;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.monitor.opentelemetry.autoconfigure.implementation.models.RemoteDependencyData;
+import com.azure.monitor.opentelemetry.autoconfigure.implementation.utils.FormattedDuration;
+
+// Casing of private fields is to match the names of fields passed down via filtering configuration
+public class DependencyDataColumns extends TelemetryColumns {
+    private String Target;
+    private long Duration; // in microseconds
+    private boolean Success;
+    private String Name;
+    private int ResultCode;
+    private String Type;
+    private String Data;
+
+
+    public DependencyDataColumns(RemoteDependencyData rdData) {
+        super();
+        setCustomDimensions(rdData.getProperties(), rdData.getMeasurements());
+        this.Target = rdData.getTarget();
+        this.Duration = FormattedDuration.getDurationFromTelemetryItemDurationString(rdData.getDuration());
+        this.Success = rdData.isSuccess();
+        this.Name = rdData.getName();
+        this.Type = rdData.getType();
+        this.Data = rdData.getData();
+        try {
+            this.ResultCode = Integer.parseInt(rdData.getResultCode());
+        } catch (NumberFormatException e) {
+            this.ResultCode = -1;
+        }
+    }
+
+    // To be used for tests only
+    public DependencyDataColumns(String target, long duration, boolean success, String name, int resultCode, String type, String data) {
+        super();
+        this.Target = target;
+        this.Duration = duration;
+        this.Success = success;
+        this.Name = name;
+        this.ResultCode = resultCode;
+        this.Type = type;
+        this.Data = data;
+    }
+
+    // To be used in tests only
+    public void setSuccess(boolean success) {
+        this.Success = success;
+    }
+
+    // To be used in tests only
+    public void setResultCode(int responseCode) {
+        this.ResultCode = responseCode;
+    }
+
+    // To be used in tests only
+    public void setDuration(long duration) {
+        this.Duration = duration;
+    }
+
+    // to be used in tests only
+    public void setData(String data) {
```

Comment:
It seems that the `setData` method only used at https://github.com/Azure/azure-sdk-for-java/blob/9d50951cdcf3e6a0e6ce8773119945c6e82a8966/sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/test/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/quickpulse/filteringTest/FilterTest.java#L298



to update a `DependencyDataColumns` instance created at https://github.com/Azure/azure-sdk-for-java/blob/9d50951cdcf3e6a0e6ce8773119945c6e82a8966/sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/test/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/quickpulse/filteringTest/FilterTest.java#L277

If so, the setter could be removed to only use a constructor, and the field could become final.

The same remark migh apply to other fields.

Reply:
You're right; the intended usage is for these fields to not change. I will fix all the fields that use these setters.

Reply:
This is fixed!